Adds a roadmap and versioning documentation (#1)
buildpacks/community: VERSIONING.md

#### Versioning in the CNB project
The CNB project has a variety of versioning schemes that contribute to overall stability and communicates the compatibility of different components.

> #### Lifecycle

Author: ### Spec versioning

    - When `<major>` is `0`, the platform is only compatible with lifecycles implementing that exact Platform API.
    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
    `<major>.<minor>`, where `<major>` of the lifecycle equals

### Buildpack API Version
Documented in the [buildpack spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpacktoml-toml) this field indicate compatibility with a given lifecycle according to the following rules:
- When <major> is 0, the buildpack is only compatible with lifecycles implementing that exact buildpack API.
- When <major> is greater than 0, the buildpack is only compatible with lifecycles implementing buildpack API <major>.<minor>, where <major> of the lifecycle equals <major> of the buildpack and <minor> of the lifecycle is greater than or equal to <minor> of the buildpack.

> #### Releases

Author: ### Implementations

# Versioning in the CNB project

> The CNB project versions our [spec components](#spec) and versions our [implementations](#implementations) following seperate schemas, detailed below:

Reviewer: Do we need to document how libbuildpack is versioned too?

Author: Good call will do.

### Spec component versioning
[spec]: #spec
The spec describes three components that are separately versioned. These APIs are versioned as follows:

#### Platform API Version
Given by CNB platforms to the lifecycle, this version indicates which platform API versions that platform currently support. In review, in [PR#67](https://github.com/buildpacks/spec/pull/67/files), this version indicates the compatibility between a CNB platform and a given lifecycle according to the following rules:
    - When `<major>` is `0`, the platform is only compatible with lifecycles implementing that exact Platform API.
    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
    `<major>.<minor>`, where `<major>` of the lifecycle equals

> #### Buildpack API Version

Reviewer: https://github.com/buildpacks/rfcs/blob/master/text/0010-api-versions.md

Documented in the [buildpack spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpacktoml-toml) this field indicate compatibility with a given lifecycle according to the following rules:
- When <major> is 0, the buildpack is only compatible with lifecycles implementing that exact buildpack API.
- When <major> is greater than 0, the buildpack is only compatible with lifecycles implementing buildpack API <major>.<minor>, where <major> of the lifecycle equals  <major> of the buildpack and <minor> of the lifecycle is greater than or equal to <minor> of the buildpack.

#### Distribution API Version

> This component is not explicitly versioned. As changes to this spec occur, an codification of the contract may be proposed in a future RFC.

Author: Should be:
> A codification

# Versioning in the CNB project

> The CNB project versions our [spec components](#spec) and versions our [implementations](#implementations) following seperate schemas, detailed below:

Reviewer 2: ```suggestion
The CNB project versions our [spec components](#spec) and [implementations](#implementations) following seperate schemas, detailed below:
```

### Spec component versioning
[spec]: #spec

> Currently the spec describes three components which are seperately versioned. These APIs are versioned as follows:

Reviewer 2: ```suggestion
The spec describes three components that are separately versioned. These APIs are versioned as follows:
```

    - When `<major>` is `0`, the platform is only compatible with lifecycles implementing that exact Platform API.
    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
    `<major>.<minor>`, where `<major>` of the lifecycle equals

#### Buildpack API Version
Documented in the [buildpack spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpacktoml-toml) this field indicate compatibility with a given lifecycle according to the following rules:
- When <major> is 0, the buildpack is only compatible with lifecycles implementing that exact buildpack API.
- When <major> is greater than 0, the buildpack is only compatible with lifecycles implementing buildpack API <major>.<minor>, where <major> of the lifecycle equals  <major> of the buildpack and <minor> of the lifecycle is greater than or equal to <minor> of the buildpack.

#### Distribution API Version

> This component is currently not explicely versioned. As changes to this spec occur, an RFC will codify the contract.

Reviewer 2: ```suggestion
This component is not explicitly versioned. As changes to this spec occur, an codification of the contract may be proposed in a future RFC.
```

### Implementation versioning
[implementations]: #implementations

> The CNB projects contains two main implementations of the spec are versioning as follows:

Reviewer 2: I'm not sure what this is trying to say

Author: Me neither. Clarified the language a bit.
